perfkitbenchmarker/publisher.py

```diff
@@ -487,8 +487,11 @@ class SampleCollector(object):
       SampleCollector._DefaultPublishers.
     run_uri: A unique tag for the run.
   """
-  def __init__(self, metadata_providers=None, publishers=None):
-    self.samples = []
+  def __init__(self, samples=None, metadata_providers=None, publishers=None):
+    if samples is None:
```

Comment:
I see. Are you explicitly trying to avoid a copy? Or expecting the
constructor caller to continue modifying their "samples" list?

On Wed, Oct 28, 2015 at 10:31 PM, skschneider <notifications@github.com>
wrote:

> In perfkitbenchmarker/publisher.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/597#discussion_r43350524>
> :
>
> > @@ -487,8 +487,11 @@ class SampleCollector(object):
> >        SampleCollector._DefaultPublishers.
> >      run_uri: A unique tag for the run.
> >    """
> > -  def __init__(self, metadata_providers=None, publishers=None):
> > -    self.samples = []
> > +  def __init__(self, samples=None, metadata_providers=None, publishers=None):
> > +    if samples is None:
>
> Changed this to
>
> self.samples = [] if samples is None else samples
>
> reason being that passing in a Falsey value like an empty list shouldn't
> create a new list.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/597/files#r43350524>
> .
>



-- 
Connor McCoy | Software Engineer | connormccoy@google.com | 206-876-1657